org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/dom/JavacTypeBinding.java

```diff
@@ -88,13 +90,15 @@ public abstract class JavacTypeBinding implements ITypeBinding {
 	public final TypeSymbol typeSymbol;
 	private final Types types;
 	private final Type type;
+	private boolean isDeclaration;
 	private boolean recovered = false;
 
-	public JavacTypeBinding(final Type type, final TypeSymbol typeSymbol, JavacBindingResolver resolver) {
+	public JavacTypeBinding(final Type type, final TypeSymbol typeSymbol, boolean isDeclaration, JavacBindingResolver resolver) {
 		if (type instanceof PackageType) {
 			throw new IllegalArgumentException("Use JavacPackageBinding");
 		}
 		this.type = type;
+		this.isDeclaration = isDeclaration;
```

Comment:
Should it only be set when `type.isParameterized()` is `true`? In other cases, it seems unnecessary to create different bindings.